
<!DOCTYPE html>
<html>

    <head>

        <meta charset="UTF-8">
        <link rel="stylesheet" href="/css/main.css">

        
    <title>PyFlowBAT</title>


        

<script type="text/javascript" id="MathJax-script" async
src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js">
</script>



    </head>

    <body>
        <header>
            

<div class="column">
    <div class="navigation">
        /
        
            
                <a href="/">About</a> /
            
        
            
                <a href="/publications">Publications</a> /
            
        
            
                <a href="/projects">Projects</a> /
            
        
    </div>
    <div class="breadcrumb">
        <a href="/">Elli Beres</a>
    </div>
</div>


        </header>
        

        

    <h1>PyFlowBAT</h1>
    <p class="subtitle">An Open-Source Software Package for Performing High-Throughput Batch Analysis of Flow Cytometry Data. Developed for the Leonard Lab. Jun 2022–Present.</p>

    <h3></h3>

    <h2 id="important"><a class="anchor" href="#important">#</a> IMPORTANT</h2>
<p>Paper writing in progress!</p>
<p><img src="logo.png" alt="PyFlowBat logo"></p>
<hr>
<h2 id="abstract"><a class="anchor" href="#abstract">#</a> Abstract</h2>
<p>Synthetic biologists frequently utilize flow cytometry to evaluate gene expression output of engineered biological systems because flow cytometry enables single-cell measurements of many samples at a time.
Although large and highly dimensional datasets can be collected using flow cytometry, analyzing said data presents a barrier to high-throughput experiments.
Currently, open-source and high-throughput flow cytometry analysis tools have not been extensively developed nor widely adopted.
The field’s reliance on closed-source, expensive, and low-throughput tools presents a steep barrier to entry and limits the potential of synthetic biology.
To address these challenges, we present the Python Flow cytometry Batch Analysis Tool (PyFlowBAT), an open-source Python package enabling rapid batch analysis of large flow cytometry datasets.
PyFlowBAT is designed to accelerate and standardize analysis workflows by performing vectorized batch operations on collections of samples instead of individually operating on each sample.
PyFlowBAT provides features for conducting the entirety of the “processing, analysis, visualization” pipeline in one application.
PyFlowBAT is both accessible to researchers with limited coding experience through its plug-and-play Jupyter Notebook templates and appropriate for advanced programmers as a library to be used in more complex programs.
We validate PyFlowBAT’s effectiveness by analyzing inducible fluorescent protein expression data in mammalian cells and comparing the quantitative metrics produced by PyFlowBAT to those produced manually using FlowJo, the field’s standard analysis tool.
PyFlowBAT adds crucial missing functionality to the growing corpus of open-source synthetic biology tools developed in recent years.
Furthermore, PyFlowBAT fills a largely ignored yet increasingly important niche within the synthetic biologist’s toolkit: software for efficient, high-throughput flow cytometry analysis and visualization.</p>
<hr>
<p>Read the poster I presented at EBRC 2023 on PyFlowBAT <a href="https://elliberes.me/publications/ebrc2023/featured.png">here</a>!
You can find the (work-in-progress) PyFlowBAT code <a href="https://github.com/leonardlab/pyflowbat">here</a>.</p>




        <footer>
            

    <hr>

    <div>Powered by
        <a href="https://golang.org/">Go</a> and
        <a href="https://github.com/ellifteria/grafe">graf&emacr;</a>
    </div>


        </footer>
    </body>

</html>




